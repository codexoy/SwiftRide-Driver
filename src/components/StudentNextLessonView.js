import React, { Fragment } from "react"
import { StyleSheet, Text, TouchableOpacity, View } from "react-native"
import Row from "./Row"
import Hours from "./Hours"
import { strings } from "../i18n"
import EmptyState from "./EmptyState"
import SimpleLoader from "./SimpleLoader"
import moment from "moment"
import { DATE_FORMAT, NAME_LENGTH } from "../consts"

export default class StudentNextLessonView extends React.Component {
	render() {
		if (this.props.loading) {
			return <SimpleLoader />
		}
		const { lesson } = this.props
		const meetup = lesson.meetup_place || strings("not_set")
		const dropoff = lesson.dropoff_place || strings("not_set")
		if (lesson) {
			return (
				<TouchableOpacity
					onPress={this.props.lessonPress}
					testID={this.props.testID}
				>
					<Fragment>
						<Row
							style={styles.lessonRow}
							leftSide={
								<View style={styles.dateView}>
									<Text style={styles.date}>
										{moment
											.utc(lesson.date)
											.local()
											.format(DATE_FORMAT)}
									</Text>
									<Hours
										duration={lesson.duration}
										date={lesson.date}
										style={styles.hoursStyle}
									/>
								</View>
							}
						>
							<Text style={styles.placeStyle}>
								{strings("teacher.new_lesson.meetup")}:{" "}
								{meetup.slice(0, NAME_LENGTH)}
							</Text>
							<Text style={styles.placeStyle}>
								{strings("teacher.new_lesson.dropoff")}:{" "}
								{dropoff.slice(0, NAME_LENGTH)}
							</Text>
						</Row>
					</Fragment>
				</TouchableOpacity>
			)
		} else {
			return (
				<EmptyState
					image="lessons"
					text={strings("empty_lessons")}
					style={styles.empty}
					imageSize="small"
				/>
			)
		}
	}
}

const styles = StyleSheet.create({
	dateView: { marginTop: -8 },
	placeStyle: { alignSelf: "flex-start" },
	date: {
		fontFamily: "Assistant-Light",
		fontSize: 24
	},
	hoursStyle: { fontWeight: "bold", alignSelf: "center" },
	lessonRow: {
		marginTop: 12
	}
})
