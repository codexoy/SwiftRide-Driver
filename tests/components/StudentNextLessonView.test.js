import React from "react"

import renderer from "react-test-renderer"

import StudentNextLessonView from "../../src/components/StudentNextLessonView"

describe("StudentNextLessonView", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<StudentNextLessonView
					lessonPress={jest.fn()}
					testID="test"
					lesson={{ duration: 40, date: new Date("2019-01-01") }}
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
