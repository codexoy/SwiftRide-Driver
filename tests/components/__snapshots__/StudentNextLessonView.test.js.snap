// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentNextLessonView view renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="test"
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 12,
        "width": "100%",
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text>
        איסוף
        : 
        לא נקבע
      </Text>
    </View>
    <View
      style={
        Object {
          "marginLeft": "auto",
        }
      }
    >
      <Text
        style={
          Object {
            "marginBottom": -8,
          }
        }
      >
        Invalid date
         - 
        Invalid date
      </Text>
    </View>
  </View>
</View>
`;
